### PR TITLE
swayosd: refactor build to install all files

### PR DESCRIPTION
**Pull request by aacebedo**
###### Description of changes

The way swayosd was build missed the systemd and dbus files. This PR use the build scripts provided by the original package.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/window-managers/sway/osd.nix [hunk 1]
```diff
@@ -21,26 +27,47 @@ rustPlatform.buildRustPackage {
...
     pkg-config
+    meson
+    rustc
+    cargo
+    ninja
+    rustPlatform.cargoSetupHook
   ];
 
   buildInputs = [
+    coreutils
```
Comment (SuperSandro2000) [inline pkgs/applications/window-managers/sway/osd.nix:47]:
```suggestion
```
that's unused and not required for postPatch to work

File: pkgs/applications/window-managers/sway/osd.nix [hunk 2]
```diff
@@ -21,26 +27,47 @@ rustPlatform.buildRustPackage {
...
+  '';
+
+  meta = with lib;
+    {
+      description = "A GTK based on screen display for keyboard shortcuts";
+      homepage = "https://github.com/ErikReider/SwayOSD";
+      license = licenses.gpl3Plus;
+      maintainers = with maintainers; [ aleksana ];
+      platforms = platforms.linux;
+    };
```
Comment (SuperSandro2000) [inline pkgs/applications/window-managers/sway/osd.nix:72]:
```suggestion
  meta = with lib; {
    description = "A GTK based on screen display for keyboard shortcuts";
    homepage = "https://github.com/ErikReider/SwayOSD";
    license = licenses.gpl3Plus;
    maintainers = with maintainers; [ aleksana ];
    platforms = platforms.linux;
  };
```

Comment (aacebedo) [reply pkgs/applications/window-managers/sway/osd.nix]: corrected

File: pkgs/applications/window-managers/sway/osd.nix (hunk 1 above)
Comment (aacebedo) [reply pkgs/applications/window-managers/sway/osd.nix:46]: corrected

**Pull request merged**